hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -585,6 +589,26 @@ public AbfsRestOperation append(final String path, final byte[] buffer,
       }
     }
 
+    if (lease != null && lease.getLeaseID() != null && !lease.getLeaseID().isEmpty()) {
```

Comment:
Add below member functions to AbfsLease : 
1. boolean hasValidLease() - which will return value of ( lease.getLeaseID() != null && !lease.getLeaseID().isEmpty())
2. AddLeaseHeaders(List<AbfsHttpHeader> requestHeaders) - and move the requestHeader conditional header add logic into it